impl/maven-impl/src/main/java/org/apache/maven/impl/model/DefaultModelInterpolator.java

```diff
@@ -101,21 +104,14 @@ interface InnerInterpolator {
     @Override
     public Model interpolateModel(
             Model model, Path projectDir, ModelBuilderRequest request, ModelProblemCollector problems) {
-        InnerInterpolator innerInterpolator = createInterpolator(model, projectDir, request, problems);
```

Comment:
coupling. 
- https://github.com/pmd/pmd/issues/5770

Reply:
I do not agree with that recommendation, and we shouldn't include the warning in our config.

Reply:
> in our config

the overall goal is to apply convention over config bumping ruleset up until we reach error-prone then we can build up from there to avoid big discussion about flavor and opinion.